submissions/pashapushak1996/js-dom/index.js

```diff
@@ -0,0 +1,190 @@
+        "location": "Earth (Replacement Dimension)",
+        "image": "https://rickandmortyapi.com/api/character/avatar/9.jpeg",
+    },
+    {
+        "id": 10,
+        "name": "Alan Rails",
+        "type": "Superhuman (Ghost trains summoner)",
+        "location": "Worldender's lair",
+        "image": "https://rickandmortyapi.com/api/character/avatar/10.jpeg",
+    },
+    {
+        "id": 11,
+        "name": "Albert Einstein",
+        "type": "",
+        "location": "Earth (Replacement Dimension)",
+        "image": "https://rickandmortyapi.com/api/character/avatar/11.jpeg",
+    },
+    {
+        "id": 12,
+        "name": "Alexander",
+        "location": "Anatomy Park",
+
+        "image": "https://rickandmortyapi.com/api/character/avatar/12.jpeg",
+    },
+    {
+        "id": 13,
+        "name": "Alien Googah",
+        "location": "Earth (Replacement Dimension)",
+        "image": "https://rickandmortyapi.com/api/character/avatar/13.jpeg",
+    },
+    {
+        "id": 14,
+        "name": "Alien Morty",
+        "location": "Citadel of Ricks",
+        "image": "https://rickandmortyapi.com/api/character/avatar/14.jpeg",
+    },
+    {
+        "id": 15,
+        "name": "Alien Rick",
+        "location": "Citadel of Ricks",
+        "image": "https://rickandmortyapi.com/api/character/avatar/15.jpeg",
+    },
+    {
+        "id": 16,
+        "name": "Amish Cyborg",
+        "location": "Earth (Replacement Dimension)",
+        "image": "https://rickandmortyapi.com/api/character/avatar/16.jpeg",
+    },
+    {
+        "id": 17,
+        "name": "Annie",
+        "location": "Anatomy Park",
+        "image": "https://rickandmortyapi.com/api/character/avatar/17.jpeg",
+    },
+    {
+        "id": 18,
+        "name": "Antenna Morty",
+        "location": "Citadel of Ricks",
+        "image": "https://rickandmortyapi.com/api/character/avatar/18.jpeg",
+    },
+    {
+        "id": 19,
+        "name": "Antenna Rick",
+        "location": "unknown",
+        "image": "https://rickandmortyapi.com/api/character/avatar/19.jpeg",
+    },
+];
+
+//On page loaded
+const loadSideMenu = () => {
+    const sideMenuList = document.querySelector('.side-menu__list');
+
+    const sideMenuListItems = characters.map((character) => {
+        const sideMenuItem = document.createElement('li');
+        sideMenuItem.className = 'side-menu__item';
+        sideMenuItem.innerHTML = `<a id="${ character.id }">${ character.name }</a>`;
+
+        return sideMenuItem;
+    });
+
+    sideMenuList.append(...sideMenuListItems);
+
+};
+
+const menuLinkOnClick = () => {
+    const sideMenuItems = document.querySelectorAll('.side-menu__item');
+
+    const characterItems = document.querySelectorAll('.character');
+    sideMenuItems.forEach((item, index) => {
+        item.addEventListener('click', toggleClass(characterItems, index, 'character_hide'));
+
+        item.addEventListener('click', toggleClass(sideMenuItems, index, 'active', 'add'));
```

Comment:
Add event listener one for all menus.
Before rendering all content add the property id to the dataset element.
After clicking, we check the link's id with dataset id content wrap.

Reply:
I have done it)